File: backend/app/models/task.py
from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey, Date, func
from sqlalchemy.orm import relationship

from app.database import Base


class TicketTask(Base):
    __tablename__ = "ticket_tasks"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    platform_account_id = Column(Integer, ForeignKey("platform_accounts.id"), nullable=False)
    show_name = Column(String(255), nullable=False)
    show_url = Column(String(500), nullable=False)
    target_date = Column(Date, nullable=True)
    sale_time = Column(DateTime, nullable=False)
    ticket_type = Column(String(100), default="")
    quantity = Column(Integer, default=1)
    status = Column(String(20), default="pending", index=True)
    priority = Column(Integer, default=5)
    extra_config_json = Column(Text, default="{}")
    scheduled_job_id = Column(String(100), default="")
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    platform_account = relationship("PlatformAccount", lazy="joined")
    executions = relationship("TaskExecution", back_populates="task", lazy="dynamic")

File: frontend/lib/models/task.dart
import 'account.dart';

class TicketTask {
  final int id;
  final int platformAccountId;
  final String showName;
  final String showUrl;
  final String? targetDate;
  final String saleTime;
  final String ticketType;
  final int quantity;
  final String status;
  final PlatformAccount? platformAccount;

  TicketTask({
    required this.id,
    required this.platformAccountId,
    required this.showName,
    required this.showUrl,
    this.targetDate,
    required this.saleTime,
    this.ticketType = '',
    this.quantity = 1,
    this.status = 'pending',
    this.platformAccount,
  });

  factory TicketTask.fromJson(Map<String, dynamic> json) {
    return TicketTask(
      id: json['id'],
      platformAccountId: json['platform_account_id'] ?? 0,
      showName: json['show_name'] ?? '',
      showUrl: json['show_url'] ?? '',
      targetDate: json['target_date'],
      saleTime: json['sale_time'] ?? '',
      ticketType: json['ticket_type'] ?? '',
      quantity: json['quantity'] ?? 1,
      status: json['status'] ?? 'pending',
      platformAccount: json['platform_account'] != null
          ? PlatformAccount.fromJson(json['platform_account'])
          : null,
    );
  }
}
